common/Geometry2d/Rect.hpp

```diff
@@ -63,6 +63,7 @@ class Rect : public Shape {
 
     Point center() const { return (pt[0] + pt[1]) / 2; }
 
+    /* Assumes that pt[0] <= pt[1] for both x and y */
```

Comment:
If we make this assumption here, would it be best to force this in the constructor?  Most Rect objects I've seen elsewhere are either defined by an upper-left `origin` along with `width` and `height` OR a top-left and bottom-right point.  It might be good if we change things with ours to swap the points in the constructor so that they're ordered.